src/generator/protocol/models.ts

```diff
@@ -18,20 +18,30 @@ export async function generateModels(session: Session<CodeModel>): Promise<strin
   // add types from requests and responses
   for (const group of values(session.model.operationGroups)) {
     for (const op of values(group.operations)) {
+      // add fields related to the operation response
       if (op.responses) {
-        // check if the response has http headers that it will expect information from. 
-        if (op.responses![0].protocol.http!.headers) {
-          for (const header of values(op.responses![0].protocol.http!.headers)) {
-            const head = <LanguageHeader>header;
-            // convert each header to a property and append it to the response properties list
-            if (!HasDescription(head)) {
-              head.description = `${head.name} contains the information returned from the ${head.name} header response.`
+        let firstResp = op.responses![0];
+        let headerArray = new Map<string, LanguageHeader>();
```

Comment:
Might be good to add a comment explaining this merging of response headers into one response type.

Reply:
Done